File: speech2textdemo/app/src/main/java/edu/cs4730/speech2textdemo/MainActivity.java
package edu.cs4730.speech2textdemo;

import androidx.activity.result.ActivityResult;
import androidx.activity.result.ActivityResultCallback;
import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.speech.RecognizerIntent;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import java.util.ArrayList;

import edu.cs4730.speech2textdemo.databinding.ActivityMainBinding;

/**
 * This is a simplified version of Google's voice recognition code.
 * It skips all the languages and just uses the default language.
 */

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "VoiceRecognition";
    ActivityMainBinding binding;
    ActivityResultLauncher<Intent> myActivityResultLauncher;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        ViewCompat.setOnApplyWindowInsetsListener(binding.main, (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return WindowInsetsCompat.CONSUMED;
        });


        binding.button1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startVoiceRecognitionActivity();
            }
        });

        //using the new startActivityForResult method.
        myActivityResultLauncher = registerForActivityResult(
            new ActivityResultContracts.StartActivityForResult(),
            new ActivityResultCallback<ActivityResult>() {
                @Override
                public void onActivityResult(ActivityResult result) {
                    if (result.getResultCode() == Activity.RESULT_OK) {
                        Intent data = result.getData();
                        // Fill the list view with the strings the recognizer thought it could have heard, there should be 5, based on the call
                        ArrayList<String> matches = data.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS);

                        //display results.
                        logthis("results: " + String.valueOf(matches.size()));
                        for (int i = 0; i < matches.size(); i++) {
                            logthis("result " + i + ":" + matches.get(i));
                        }
                    }
                }
            });

    }

    /**
     * Fire an intent to start the speech recognition activity.
     */
    private void startVoiceRecognitionActivity() {
        Intent intent = new Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH);

        // Specify the calling package to identify your application
        intent.putExtra(RecognizerIntent.EXTRA_CALLING_PACKAGE, getClass().getPackage().getName());

        // Display an hint to the user about what he should say.
        intent.putExtra(RecognizerIntent.EXTRA_PROMPT, "Say Something!");

        // Given an hint to the recognizer about what the user is going to say
        intent.putExtra(RecognizerIntent.EXTRA_LANGUAGE_MODEL,
            RecognizerIntent.LANGUAGE_MODEL_FREE_FORM);

        // Specify how many results you want to receive. The results will be sorted
        // where the first result is the one with higher confidence.
        intent.putExtra(RecognizerIntent.EXTRA_MAX_RESULTS, 5);

        Log.i(TAG, "Calling the Voice Intenet");
        myActivityResultLauncher.launch(intent);
    }

    /**
     * simple method to add the log TextView and debug log.
     */
    public void logthis(String newinfo) {
        Log.d(TAG, newinfo);
        binding.logger.append(newinfo + "\n");
    }
}

File: speech2textdemo_kt/app/src/main/java/edu/cs4730/speech2textdemo_kt/MainActivity.kt
package edu.cs4730.speech2textdemo_kt

import android.content.Intent
import android.os.Bundle
import android.speech.RecognizerIntent
import android.util.Log
import androidx.activity.result.ActivityResult
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity
import edu.cs4730.speech2textdemo_kt.databinding.ActivityMainBinding


/**
 * This is a simplified version of Google's voice recognition code.
 * It skips all the languages and just uses the default language.
 */
class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private lateinit var myActivityResultLauncher: ActivityResultLauncher<Intent>
    companion object {
        private const val TAG = "VoiceRecognition"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.getRoot())
        binding.button1.setOnClickListener { startVoiceRecognitionActivity() }

        //using the new startActivityForResult method.
        registerForActivityResult<Intent, ActivityResult>(
            ActivityResultContracts.StartActivityForResult()
        ) { result ->
            if (result.resultCode == RESULT_OK) {
                val data = result.data
                // Fill the list view with the strings the recognizer thought it could have heard, there should be 5, based on the call
                val matches = data!!.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS)

                //display results.
                logthis("results: " + matches!!.size.toString())
                for (i in matches.indices) {
                    logthis("result " + i + ":" + matches[i])
                }
            }
        }.also { myActivityResultLauncher = it }
    }

    /**
     * Fire an intent to start the speech recognition activity.
     */
    private fun startVoiceRecognitionActivity() {
        val intent = Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH)

        // Specify the calling package to identify your application
        intent.putExtra(RecognizerIntent.EXTRA_CALLING_PACKAGE, javaClass.getPackage()?.name)

        // Display an hint to the user about what he should say.
        intent.putExtra(RecognizerIntent.EXTRA_PROMPT, "Say Something!")

        // Given an hint to the recognizer about what the user is going to say
        intent.putExtra(
            RecognizerIntent.EXTRA_LANGUAGE_MODEL,
            RecognizerIntent.LANGUAGE_MODEL_FREE_FORM
        )

        // Specify how many results you want to receive. The results will be sorted
        // where the first result is the one with higher confidence.
        intent.putExtra(RecognizerIntent.EXTRA_MAX_RESULTS, 5)

        Log.i(TAG, "Calling the Voice Intenet")
        myActivityResultLauncher.launch(intent)
    }

    /**
     * simple method to add the log TextView and debug log.
     */
    private fun logthis(newinfo: String) {
        Log.d(TAG, newinfo)
        binding.logger.append(newinfo + "\n")
    }


}
